tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -2039,8 +2041,10 @@ SpirvInstruction *SpirvEmitter::processCall(const CallExpr *callExpr) {
       argInfo = declIdMapper.getDeclEvalInfo(declRefExpr->getDecl());
     }
 
-    if (argInfo && argInfo->getStorageClass() == spv::StorageClass::Function &&
-        canActAsOutParmVar(param) &&
+    if (argInfo &&
+        ((argInfo->getStorageClass() == spv::StorageClass::Function &&
+          canActAsOutParmVar(param)) ||
+         argInfo->getStorageClass() == spv::StorageClass::Workgroup) &&
```

Comment:
What about variables in the Private storage class? (e.g. static variable)

again, I think with the new SPIRV-Tools behavior, we shouldn't create temporary variables unless we have an rvalue? Also think about the cases where the function parameter is `out` or `inout`.